### ISSUE TITLE
COEFF_TO_PTR calls are flagged by -fsanitize=undefined

### ISSUE THREAD

**Issue by saraedum** [wbhart/flint2#991]
`COEFF_TO_PTR` performs a left shift `<< 2` on an `fmpz`, i.e., an `slong`.

This operation is flagged by compilers with `-fsanitize=undefined`:

```
flint/fmpz.h:457:26: runtime error: left shift of 4611709672739175420 by 2 places cannot be represented in type 'long int'
```

According to the C standard this is indeed undefined behavior (shifting into the sign bit):
![image](https://user-images.githubusercontent.com/373765/130063630-e6db1b19-b6ab-4ca0-a53f-0f008a8f8313.png)
